src/main/java/edu/iu/terracotta/service/app/impl/AssessmentServiceImpl.java

```diff
@@ -263,6 +256,24 @@ public AssessmentDto toDto(Assessment assessment, Long submissionId, boolean que
         return assessmentDto;
     }
 
+    private List<QuestionDto> handleQuestionDtos(Assessment assessment, Long submissionId, boolean showQuestions, boolean showAnswers, boolean isStudent) {
+        if (!showQuestions) {
+            return Collections.emptyList();
+        }
+
+        AtomicBoolean showCorrectAnswers = new AtomicBoolean(true);
+
+        if (isStudent) {
+            // is a student, use assessment setting
+            showCorrectAnswers.set(assessment.canViewCorrectAnswers());
+        }
```

Comment:
If isStudent is true, then showCorrectAnswers should be false, always.

Reply:
Oh, so this endpoint isn't used in the display of correct answers? I'll fix it.

Reply:
The reason for the fix was Ryan was saying that he needed the "correct" field to be set...